### PR TITLE
Logical delete schemas

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java [hunk 1]
```diff
@@ -187,11 +194,39 @@ public void test() throws Exception {
         assertTrue(foundRevSet.contains(1L));
         assertTrue(foundRevSet.contains(2L));
         assertTrue(foundRevSet.contains(3L));
+        
+        UploadSchemaList schemasList = devUploadSchemasApi.getMostRecentUploadSchemas(true).execute().body();
+        List<Long> firstRevisions = schemasList.getItems().stream().map(UploadSchema::getRevision)
+                .collect(Collectors.toList());
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java]: Should this be moved to after the delete so you can verify that the includeDeleted flag actually includes deleted?

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java [hunk 2]
```diff
@@ -187,11 +194,39 @@ public void test() throws Exception {
...
+        // All are returned if deleted are included
+        schemasList = devUploadSchemasApi.getAllRevisionsOfUploadSchema(schemaId, true).execute().body();
+        assertEquals(3, schemasList.getItems().size());
+        
+        // This still works, despite the fact the schema is logically deleted
+        UploadSchema aSchema = devUploadSchemasApi.getMostRecentUploadSchema(schemaId).execute().body();
+        assertTrue(aSchema.isDeleted());
+        
+        // Physically delete all the schemas
+        adminApi.deleteAllRevisionsOfUploadSchema(schemaId, true).execute();
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java]: Is it worth testing the logical delete of this schema first?

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java [hunk 3]
```diff
@@ -187,11 +194,39 @@ public void test() throws Exception {
...
         assertTrue(foundRevSet.contains(2L));
         assertTrue(foundRevSet.contains(3L));
+        
+        UploadSchemaList schemasList = devUploadSchemasApi.getMostRecentUploadSchemas(true).execute().body();
+        List<Long> firstRevisions = schemasList.getItems().stream().map(UploadSchema::getRevision)
+                .collect(Collectors.toList());
+        
+        // Step 5: Logically delete one of the schemas with the test schema ID
+        Long mostRecentRev = devUploadSchemasApi.getMostRecentUploadSchema(schemaId).execute().body().getRevision();
+        adminApi.deleteUploadSchema(schemaId, mostRecentRev, false).execute();
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java]: Is it worth also testing a physical delete of the schema rev?

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java [hunk 4]
```diff
@@ -187,11 +194,39 @@ public void test() throws Exception {
...
+        
+        // This will return different version numbers, as more recent versions are deleted
+        schemasList = devUploadSchemasApi.getMostRecentUploadSchemas(false).execute().body();
+        List<Long> secondRevisions = schemasList.getItems().stream().map(UploadSchema::getRevision)
+                .collect(Collectors.toList());
 
-        // Step 5: Delete all schemas with the test schema ID
-        adminApi.deleteAllRevisionsOfUploadSchema(IntegTestUtils.STUDY_ID, schemaId).execute();
+        // Not equal because different revisions are now returned
+        assertNotEquals(firstRevisions, secondRevisions);
```
Comment (DwayneJengSage) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java]: Should we verify that firstRevisions includes deleted revisions while secondRevisions doesn't?

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java (hunk 2 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java]: I feel like line 200 (where we logically delete) to line 226 test the logically deleted schema. What else should be tested there?

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java (hunk 3 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java]: Added this right before deleting all revisions.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java (hunk 4 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/UploadSchemaTest.java]: Okay I changed this to directly test for the presence/absence of mostRecentRev

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**